pandas/io/formats/info.py

```diff
@@ -209,147 +250,355 @@ def info(self) -> None:
         --------
         %(examples_sub)s
         """
-        lines = []
+        printer = InfoPrinter(
```

Comment:
Hmm at first glance I think this delegation pattern is a little confusing but maybe I am overlooking something. What is the advantage to calling `DataFrameInfo.to_buffer` from `.info` instead of calling `InfoPrinter.to_buffer` instead?

Reply:
I call ``DataFrameInfo.to_buffer`` from ``DataFrame`` because of the following reasons.
1. Mainly did not want to mess with the docstring (which is defined in ``DataFrameInfo.to_buffer``)
2. I wanted to avoid import of ``InfoPrinter`` into ``pandas/core/frame.py``
3. Probably it is a good idea to make public API function as disconnected from the internal API as possible. Right now changes inside ``pandas/io/formats/io.py`` will have slight effect on ``DataFrame`` class, since the info printing functionality is encapsulated inside ``DataFrameInfo``. If, however, we call ``InfoPrinter.to_buffer`` from ``DataFrame.info()``, then we would need to ensure that all the possible changes made inside ``InfoPrinter`` and ``DataFrameInfo`` are properly communicated to ``DataFrame.info``. I hope it makes sense.

But I do not strongly object against having ``InfoPrinter.to_buffer`` called from ``DataFrame.info()``.